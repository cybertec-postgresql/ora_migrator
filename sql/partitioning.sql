/* for requirements, see "migrate.sql" */

/* connect as migration user */
\connect - migrator

SET client_min_messages = WARNING;
SET datestyle = 'ISO, MDY';

/* set up staging schemas */
SELECT db_migrate_prepare(
   plugin => 'ora_migrator',
   server => 'oracle',
   only_schemas => ARRAY['TESTSCHEMA3'],
   options => JSONB '{"max_long": 1024}'
);

/* create foreign tables */
SELECT db_migrate_mkforeign(
   plugin => 'ora_migrator',
   server => 'oracle',
   options => JSONB '{"max_long": 1024}'
);

/* migrate the data */
SELECT db_migrate_tables(
   plugin => 'ora_migrator'
);

/* migrate the constraints */
SELECT db_migrate_constraints(
   plugin => 'ora_migrator'
);

/* clean up */
SELECT db_migrate_finish();

/* check results */

\d+ testschema3.part1
\d+ testschema3.part2
\d+ testschema3.part3

SELECT tableoid::regclass partname, * FROM testschema3.part1;
SELECT tableoid::regclass partname, * FROM testschema3.part2;
SELECT tableoid::regclass partname, * FROM testschema3.part3;
