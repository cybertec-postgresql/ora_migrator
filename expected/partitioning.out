/* for requirements, see "migrate.sql" */
/* connect as migration user */
\connect - migrator
SET client_min_messages = WARNING;
SET datestyle = 'ISO, MDY';
/* set up staging schemas */
SELECT db_migrate_prepare(
   plugin => 'ora_migrator',
   server => 'oracle',
   only_schemas => ARRAY['TESTSCHEMA3'],
   options => JSONB '{"max_long": 1024}'
);
 db_migrate_prepare 
--------------------
                  0
(1 row)

/* create foreign tables */
SELECT db_migrate_mkforeign(
   plugin => 'ora_migrator',
   server => 'oracle',
   options => JSONB '{"max_long": 1024}'
);
 db_migrate_mkforeign 
----------------------
                    0
(1 row)

/* migrate the data */
SELECT db_migrate_tables(
   plugin => 'ora_migrator'
);
 db_migrate_tables 
-------------------
                 0
(1 row)

/* migrate the constraints */
SELECT db_migrate_constraints(
   plugin => 'ora_migrator'
);
 db_migrate_constraints 
------------------------
                      0
(1 row)

/* clean up */
SELECT db_migrate_finish();
 db_migrate_finish 
-------------------
                 0
(1 row)

/* check results */
\d+ testschema3.part1
                                  Partitioned table "testschema3.part1"
 Column |          Type          | Collation | Nullable | Default | Storage  | Stats target | Description 
--------+------------------------+-----------+----------+---------+----------+--------------+-------------
 c1     | integer                |           | not null |         | plain    |              | 
 c2     | character varying(100) |           | not null |         | extended |              | 
Partition key: LIST (c1)
Partitions: testschema3.part1_a FOR VALUES IN (1, 2, 3),
            testschema3.part1_b FOR VALUES IN (4, 5),
            testschema3.part1_default DEFAULT

\d+ testschema3.part2
                                      Partitioned table "testschema3.part2"
 Column |              Type              | Collation | Nullable | Default | Storage  | Stats target | Description 
--------+--------------------------------+-----------+----------+---------+----------+--------------+-------------
 r1     | integer                        |           | not null |         | plain    |              | 
 h1     | character varying(100)         |           | not null |         | extended |              | 
 h2     | timestamp(6) without time zone |           | not null |         | plain    |              | 
Partition key: RANGE (r1)
Partitions: testschema3.part2_a FOR VALUES FROM (MINVALUE) TO (10), PARTITIONED,
            testschema3.part2_b FOR VALUES FROM (10) TO (100), PARTITIONED,
            testschema3.part2_c FOR VALUES FROM (100) TO (MAXVALUE), PARTITIONED

\d+ testschema3.part3
                                  Partitioned table "testschema3.part3"
 Column |          Type          | Collation | Nullable | Default | Storage  | Stats target | Description 
--------+------------------------+-----------+----------+---------+----------+--------------+-------------
 c1     | integer                |           |          |         | plain    |              | 
 c2     | character varying(100) |           |          |         | extended |              | 
Partition key: HASH (c2)
Partitions: testschema3.part3_a FOR VALUES WITH (modulus 3, remainder 0), PARTITIONED,
            testschema3.part3_b FOR VALUES WITH (modulus 3, remainder 1), PARTITIONED,
            testschema3.part3_c FOR VALUES WITH (modulus 3, remainder 2), PARTITIONED

SELECT tableoid::regclass partname, * FROM testschema3.part1;
         partname          | c1 |  c2  
---------------------------+----+------
 testschema3.part1_a       |  1 | one
 testschema3.part1_b       |  5 | five
 testschema3.part1_default | 10 | ten
(3 rows)

SELECT tableoid::regclass partname, * FROM testschema3.part2;
       partname        | r1  |    h1     |         h2          
-----------------------+-----+-----------+---------------------
 testschema3.part2_a_2 |   1 | other     | 2023-01-01 23:00:00
 testschema3.part2_a_2 |   1 | something | 2023-01-01 13:00:00
 testschema3.part2_b_2 |  50 | other     | 2023-01-01 23:00:00
 testschema3.part2_b_2 |  50 | something | 2023-01-01 13:00:00
 testschema3.part2_c_2 | 500 | other     | 2023-01-01 23:00:00
 testschema3.part2_c_2 | 500 | something | 2023-01-01 13:00:00
(6 rows)

SELECT tableoid::regclass partname, * FROM testschema3.part3;
       partname        | c1 | c2 
-----------------------+----+----
 testschema3.part3_a_1 |  1 | b
 testschema3.part3_a_1 |  1 | d
 testschema3.part3_a_1 |  1 | e
 testschema3.part3_a_1 |  1 | f
 testschema3.part3_a_2 |  5 | b
 testschema3.part3_a_2 |  5 | d
 testschema3.part3_a_2 |  5 | e
 testschema3.part3_a_2 |  5 | f
 testschema3.part3_a_3 |  9 | b
 testschema3.part3_a_3 |  9 | d
 testschema3.part3_a_3 |  9 | e
 testschema3.part3_a_3 |  9 | f
 testschema3.part3_c_1 |  1 | c
 testschema3.part3_c_1 |  1 | a
 testschema3.part3_c_2 |  5 | c
 testschema3.part3_c_2 |  5 | a
 testschema3.part3_c_3 |  9 | c
 testschema3.part3_c_3 |  9 | a
(18 rows)

